packages/server-express/src/middlewares/session-middleware.ts

```diff
@@ -0,0 +1,92 @@
+import type { TSessionMiddlewareOptions } from '../types';
+import type { Request, Response, NextFunction } from 'express';
+
+import { CLOUD_IDENTIFIERS } from '../constants';
+import expressJwtMiddleware from 'express-jwt';
+import jwksRsa from 'jwks-rsa';
+
+const decodedTokenKey = 'decoded_token';
+
+const mapCloudIdentifierToUrl = (
+  issuer: TSessionMiddlewareOptions['issuer']
+) => {
+  switch (issuer) {
+    case CLOUD_IDENTIFIERS['gcp-au']:
+      return 'https://mc-api.australia-southeast1.gcp.commercetools.com';
+    case CLOUD_IDENTIFIERS['gcp-eu']: {
+      return 'https://mc-api.europe-west1.gcp.commercetools.com';
+    }
+    case CLOUD_IDENTIFIERS['gcp-us']: {
+      return 'https://mc-api.us-central1.gcp.commercetools.com';
+    }
+    case CLOUD_IDENTIFIERS['aws-fra']:
+      return 'https://mc-api.eu-central-1.aws.commercetools.com';
+    case CLOUD_IDENTIFIERS['aws-ohio']:
+      return 'https://mc-api.us-east-2.aws.commercetools.com';
+    default:
+      return undefined;
+  }
+};
+
+type TDecodedJWT = {
+  sub: string;
+  iss: string;
+  [property: string]: string;
+};
+
+const writeSessionContext = (
+  request: Request & { decoded_token: TDecodedJWT }
+) => {
+  const decodedToken = request[decodedTokenKey];
+  const publicClaimForProjectKey = `${decodedToken.iss}/claims/project_key`;
+
+  // @ts-ignore
+  request.session = {
+    userId: decodedToken.sub,
+    projectKey: decodedToken[publicClaimForProjectKey],
+  };
+
+  delete request.decoded_token;
+};
+
+function createSessionMiddleware(options: TSessionMiddlewareOptions) {
+  const configuredIssuer =
+    mapCloudIdentifierToUrl(options.issuer) ?? options.issuer;
+
+  // Return the middleware
+  return (request: Request, response: Response, next: NextFunction) => {
+    let issuer = configuredIssuer;
+    if (options.inferIssuerFromCustomHeader) {
+      // Attempt to infer the
+      const cloudIdentifierHeader = request.header('x-mc-api-cloud-identifier');
+      if (cloudIdentifierHeader) {
+        issuer = mapCloudIdentifierToUrl(cloudIdentifierHeader) ?? issuer;
+      }
+    }
```

Comment:
```suggestion
    const cloudIdentifierHeader = request.header('x-mc-api-cloud-identifier');
    const overwriteDefaultIssuer = options.inferIssuerFromCustomHeader && cloudIdentifierHeader
    const issuer = overwriteDefaultIssuer ? mapCloudIdentifierToUrl(cloudIdentifierHeader) ?? issuer : configuredDefaultIssuer;
```

Would help be get around this if nesting maybe somehow.

Reply:
Hmm I find the nesting more explicit and easier to read. Also it helps TS to infer things better.